rfcs/007 - Alternative Ways to Create Mappers.md

```diff
@@ -0,0 +1,122 @@
+# 007 - Alternative Ways to Create Mappers
+
+## Current Status
+
+### Proposed
+
+2023-02-07
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- Full Name <email@example.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Mohammad Teimori Pabandi <m@iasql.com>
+
+## Summary
+
+Having Pure-SQL modules would help us expand more. One key advantage is that we don't need a PR on our engine to add coverage for a new service, we just use a syntax like this and everything JustWorks™.
+```postgresql
+SELECT iasql_install('https://github.com/someModuleDeveloper/awsEc2Module');
+```
+That's where we started, but it might be useful to assess different paths we might take for creation of an IaSQL module/mapper.
+
+## Proposal
+
+To create a module, we need a mapper. I'll continue this RFC with my proposed solution for mappers, since a module is a set of mappers, and entities (with a bit of oversimplification).
+
+First, let's see what we need to create a mapper:
+- A cloud id function
+- An equality checker function
+- An entity, that the mapper manages it
```

Comment:
The conclusion we came to earlier and that @aguillenv is working on to have 1 mapper per entity *ought* to make equality-checking a potentially automate-able behavior, then this would not need to be defined by hand.

We already have it for some entities, but we may be able to subsume it entirely into the engine.

Reply:
The only issue that comes to my mind regarding this, is about FKs. Because we have some logics there.

Reply:
True, but we can inspect the schema of tables in Postgres itself using the [information_schema](https://stackoverflow.com/questions/1152260/how-to-list-table-foreign-keys#1152321) namespace and tables, so we *should* be able to replicate any meta-programming we're doing in Typescript in pure Postgres.

Reply:
One new thing here is for a situation where we're using `policiesAreEqual`. In that case, the exact content of two `policy` fields may not be equal, but they in-fact might be the same thing. Automating that might be a tricky thing. Or we'd need to provide a "customized" behavior for that case.